File: src/main/java/me/arasple/mc/trhologram/api/events/HologramInteractEvent.java
package me.arasple.mc.trhologram.api.events;

import me.arasple.mc.trhologram.hologram.Hologram;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.bukkit.event.player.PlayerEvent;

/**
 * @author Arasple
 * @date 2020/1/29 22:20
 */
public class HologramInteractEvent extends PlayerEvent {


    private static final HandlerList HANDLERS = new HandlerList();
    private Hologram hologram;

    public HologramInteractEvent(Player who, Hologram hologram) {
        super(who);
        this.hologram = hologram;
    }

    public static HandlerList getHandlerList() {
        return HANDLERS;
    }

    public void call() {
        Bukkit.getPluginManager().callEvent(this);
    }

    public Hologram getHologram() {
        return hologram;
    }

    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

}

File: src/main/kotlin/me/arasple/mc/trhologram/api/event/HologramInteractEvent.kt
package me.arasple.mc.trhologram.api.event

import io.izzel.taboolib.module.event.EventCancellable
import io.izzel.taboolib.module.event.EventNormal
import me.arasple.mc.trhologram.module.display.Hologram
import org.bukkit.Bukkit
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2021/2/11 16:34
 */
class HologramInteractEvent(val player: Player, val type: Type, val hologram: Hologram) : EventCancellable<HologramInteractEvent>() {

    init {
        async(!Bukkit.isPrimaryThread())
    }

    enum class Type {

        ALL,

        LEFT,

        RIGHT,

        SHIFT_LEFT,

        SHIFT_RIGHT

    }

}